jvm/src/main/kotlin/app/cash/trifle/Trifle.kt

```diff
@@ -48,21 +59,38 @@ object Trifle {
       }
     }
 
-    // Extract the x.509 certificate leaf from certificate list passed in
-    val certHolder = X509CertificateHolder(certificatesIn[0].certificate?.toByteArray())
+    val cf = CertificateFactory.getInstance("X.509");
+    val root: Certificate? = cf.generateCertificate(ByteArrayInputStream(certAnchor))
+    val list = ArrayList<Certificate>()
+
+    // Set up certPath. Remove root if in certPath
+    for(i in 1..certificatesIn.size) {
+      val cert = cf.generateCertificate( ByteArrayInputStream(certificatesIn[i-1].certificate?.toByteArray() ))
+      list.add(cert)
+    }
+    list.remove(root)
+    val certPath = cf.generateCertPath(list)
+
+    // set up trust anchor to validate certPath
+    val anchor = TrustAnchor(root as X509Certificate, null)
+    val params = PKIXParameters(setOf(anchor))
+    params.isRevocationEnabled = false
+
+    val pathValidator = CertPathValidator.getInstance(CertPathValidator.getDefaultType())
+    pathValidator.validate(certPath, params)
+
+    // Verify the signature
+    // get the public key needed to verify signature
+    val certHolder = X509CertificateHolder(certificatesIn.first().certificate?.toByteArray())
     val publicKey = KeyFactory.getInstance("EC")
       .generatePublic(
```

Comment:
shouldn't this depend on the algorithmId + public key present below?

Reply:
https://ccp-cashapp.atlassian.net/browse/PRODSQUAD-619

Reply:
Can you elaborate please?